OpenRA.Mods.Common/Widgets/Logic/Ingame/LoadIngamePlayerOrObserverUILogic.cs

```diff
@@ -46,9 +47,19 @@ public LoadIngamePlayerOrObserverUILogic(Widget widget, World world)
 
 			world.GameOver += () =>
 			{
-				worldRoot.RemoveChildren();
+				Ui.CloseWindow();
 				menuRoot.RemoveChildren();
-				Game.LoadWidget(world, "LEAVE_MAP_WIDGET", menuRoot, new WidgetArgs());
+
+				if (world.LocalPlayer != null)
+				{
+					var video = world.LocalPlayer.WinState == WinState.Won ? world.Map.Videos.GameWon : world.Map.Videos.GameLost;
+					if (!string.IsNullOrEmpty(video))
+						Media.PlayFMVFullscreen(world, video, () => { });
```

Comment:
I think this could also be done in Media if it implemented IGameOver interface. Should I move this to Media.GameOver()?